                                        PYTHON PROGRAMMING

                                    MTN COTE D'IVOIRE DATA ANALYSIS

**RESEARCH OBJECTIVE**

Currently MTN Cote d'Ivoire would like to upgrade its technology infrastructure for its mobile users in Ivory Coast. Studying the given dataset, how does MTN Cote d'Ivoire go about the upgrade of its infrastructure strategy within the given cities?

In [125]:
#we begin by first importing pandas and numpy
import pandas as pd
import numpy as np


IMPORTING THE DATA TABLES

In [126]:
#loadinng our data tables
# CRD Decription table
cdr = pd.read_excel('CDR_description.xlsx')
cdr = pd.DataFrame(cdr)
cdr

Column name                                        Description   Format
0          PRODUCT                                       Voice or SMS   String
1            VALUE                                      Billing price  Integer
2        DATE_TIME               Time in format yyyy-MM-dd hh:mm:ss.0   String
3     CELL_ON_SITE  Which cell in the site was used (not needed here)  Integer
4  DW_A_NUMBER_INT  Anonymized phone number of the person for whic...   String
5  DW_B_NUMBER_INT        Anonymized phone number of the counterparty   String
6        COUNTRY_A                  Country of party A (useless here)   String
7        COUNTRY_B                  Country of party B (useless here)   String
8          CELL_ID                                     ID of the cell   String
9          SITE_ID                                     ID of the SITE   String

In [127]:
#cells geo description table
cellgeodesc = pd.read_excel('cells_geo_description.xlsx')
cellgeodesc = pd.DataFrame(cellgeodesc)
cellgeodesc

Column name                                        Description  Format
0         VILLES                                               City  String
1         STATUS                                  In Service or not  String
2   LOCALISATION                                  in ABIDJAN or not  String
3     DECOUPZONE                                  Geographical Zone  String
4       ZONENAME                                       Name of Zone  String
5      LONGITUDE                                          Longitude   Float
6       LATITUDE                                           Latitude   Float
7         REGION                                             Region  String
8           AREA                                               Area  String
9        CELL_ID                                     ID of the cell  String
10     SITE_CODE  Site (there are several cells per site, severa...  String

In [128]:
#cells geo table
cellgeo = pd.read_csv('cells_geo.csv', delimiter = r';', engine ='python')
cellgeo = pd.DataFrame(cellgeo)
cellgeo

Unnamed: 0    VILLES      STATUS LOCALISATION    DECOUPZONE  \
0              0    ADJAME  In Service      ABIDJAN  "Abidjan_EST   
1              1    ADJAME  In Service      ABIDJAN  "Abidjan_EST   
2              2    ADJAME  In Service      ABIDJAN  "Abidjan_EST   
3              3   ASSINIE  In Service    INTERIEUR     Grand-EST   
4              4   ASSINIE  In Service    INTERIEUR     Grand-EST   
...          ...       ...         ...          ...           ...   
3969        3969  ZUENOULA  In Service    INTERIEUR    Grand-NORD   
3970        3970  ZUENOULA  In Service    INTERIEUR    Grand-NORD   
3971        3971  ZUENOULA  In Service    INTERIEUR    Grand-NORD   
3972        3972  ZUENOULA  In Service    INTERIEUR    Grand-NORD   
3973        3973  ZUENOULA  In Service    INTERIEUR    Grand-NORD   

                 ZONENAME  LONGITUDE  LATITUDE     REGION          AREA  \
0                AGBANNEW  -4.000667  5.368194     LAGUNE           NaN   
1                AGBANNEW  -4.000667  5.368194     LAGUNE           NaN   
2                AGBANNEW  -4.000667  5.368194     LAGUNE           NaN   
3                ASSINIE"  -3.488467  5.158761  SUD COMOE           NaN   
4                ASSINIE"  -3.488467  5.158761  SUD COMOE           NaN   
...                   ...        ...       ...        ...           ...   
3969  ZONE A - Grand-NORD  -6.039694  7.428611   MARAHOUE  CENTRE NORTH   
3970  ZONE A - Grand-NORD  -6.039694  7.428611   MARAHOUE  CENTRE NORTH   
3971  ZONE A - Grand-NORD  -6.065450  7.444592   MARAHOUE  CENTRE NORTH   
3972  ZONE A - Grand-NORD  -6.065450  7.444592   MARAHOUE  CENTRE NORTH   
3973  ZONE A - Grand-NORD  -6.065450  7.444592   MARAHOUE  CENTRE NORTH   

         CELL_ID   SITE_CODE  
0     fbb2e75af3  0bb492ef17  
1     1502501d78  0bb492ef17  
2     0c2e182a0c  0bb492ef17  
3     69bdfc69fd  8a1e6ce31f  
4     7a1029d2ab  8a1e6ce31f  
...          ...         ...  
3969  1cfc2d7ab1  f91ed8d2e9  
3970  3ae1708153  f91ed8d2e9  
3971  cb71f8a29c  61bee9c70c  
3972  d4a6a0b645  61bee9c70c  
3973  f9f3e42589  61bee9c70c  

[3974 rows x 12 columns]

In [129]:
Telcom1 = pd.read_csv('Telcom_dataset.csv')
Telcom1 = pd.DataFrame(Telcom1)
Telcom1

PRODUTC  VALUE               DATETIME CELL_ON_SITE  DW_A_NUMBER_INT  \
0      Voice      0  2012-05-06 23:04:37.0            1  49fae048b547780   
1        sms      0  2012-05-06 23:05:25.0            0  7a54a6eaee19e65   
2        sms     25  2012-05-06 23:05:41.0            0  6008887b0207d82   
3      Voice     56  2012-05-06 23:05:42.0            1  7370032e20f4e96   
4      Voice     14  2012-05-06 23:06:29.0            1  966ba4656e38e62   
...      ...    ...                    ...          ...              ...   
4996     sms    100  2012-05-07 00:01:54.0            0  92e367d671f45c5   
4997   Voice      1  2012-05-07 00:01:54.0            1  83b44b18238e602   
4998   Voice      0  2012-05-07 00:01:54.0            1  49fae048b547780   
4999   Voice      0  2012-05-07 00:01:54.0            0  a1613f1a992716a   
5000   Voice      7  2012-05-07 00:01:54.0            2  1670a5f7c22f4ed   

      DW_B_NUMBER_INT COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  
0     1a49f8b0ded9de6     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  
1     93343ba6e41af95     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41  
2     7aacc9709ce0a20     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi  
3     6af4a60e74c91d3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3  
4     02aa736ee555d5c     19e2e     19e2e  051KhYDCpv  051KhYDCpv  
...               ...       ...       ...         ...         ...  
4996  38811081cf82c90     19e2e     OTHER  0c2371d9bc  1e751d08a4  
4997  6ec63d4aa5de5dd     19e2e     19e2e  5fb791cf56  65efe2bc62  
4998  1a49f8b0ded9de6     19e2e     19e2e  7773627c12  9ef9616350  
4999  bd663f07b6c8bcd     19e2e     19e2e  3f4af49d6c  969b8d0e39  
5000  3600efd094f2225     19e2e     19e2e  6f4d9b77a5  ebfc58af70  

[5001 rows x 10 columns]

In [130]:
#Updating the column names to make the uniform aand correct tyoing errors
Telcom1.columns =['PRODUCT','VALUE','DATE_TIME','CELL_ON_SITE','DW_A_NUMBER','DW_B_NUMBER','COUNTRY_A','COUNTRY_B','CELL_ID','SITE_ID']
Telcom1

PRODUCT  VALUE              DATE_TIME CELL_ON_SITE      DW_A_NUMBER  \
0      Voice      0  2012-05-06 23:04:37.0            1  49fae048b547780   
1        sms      0  2012-05-06 23:05:25.0            0  7a54a6eaee19e65   
2        sms     25  2012-05-06 23:05:41.0            0  6008887b0207d82   
3      Voice     56  2012-05-06 23:05:42.0            1  7370032e20f4e96   
4      Voice     14  2012-05-06 23:06:29.0            1  966ba4656e38e62   
...      ...    ...                    ...          ...              ...   
4996     sms    100  2012-05-07 00:01:54.0            0  92e367d671f45c5   
4997   Voice      1  2012-05-07 00:01:54.0            1  83b44b18238e602   
4998   Voice      0  2012-05-07 00:01:54.0            1  49fae048b547780   
4999   Voice      0  2012-05-07 00:01:54.0            0  a1613f1a992716a   
5000   Voice      7  2012-05-07 00:01:54.0            2  1670a5f7c22f4ed   

          DW_B_NUMBER COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  
0     1a49f8b0ded9de6     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  
1     93343ba6e41af95     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41  
2     7aacc9709ce0a20     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi  
3     6af4a60e74c91d3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3  
4     02aa736ee555d5c     19e2e     19e2e  051KhYDCpv  051KhYDCpv  
...               ...       ...       ...         ...         ...  
4996  38811081cf82c90     19e2e     OTHER  0c2371d9bc  1e751d08a4  
4997  6ec63d4aa5de5dd     19e2e     19e2e  5fb791cf56  65efe2bc62  
4998  1a49f8b0ded9de6     19e2e     19e2e  7773627c12  9ef9616350  
4999  bd663f07b6c8bcd     19e2e     19e2e  3f4af49d6c  969b8d0e39  
5000  3600efd094f2225     19e2e     19e2e  6f4d9b77a5  ebfc58af70  

[5001 rows x 10 columns]

In [131]:
Telcom2 = pd.read_csv('Telcom_dataset2.csv')
Telcom2 = pd.DataFrame(Telcom2)
Telcom2

PRODUCT  VALUE              DATE_TIME CELL_ON_SITE      DW_A_NUMBER  \
0        sms      0  2012-05-07 23:02:06.0            2  32255675c3b6fe3   
1      Voice      0  2012-05-07 23:03:44.0            1  ff16d6ac74c1c64   
2        sms      0  2012-05-07 23:04:06.0            0  98d9f2fb5ff624b   
3      Voice     37  2012-05-07 23:04:48.0            1  3f9c8a33ec7be1f   
4        sms     25  2012-05-07 23:04:58.0            1  7bcccc03adf3864   
...      ...    ...                    ...          ...              ...   
4996     sms      0  2012-05-08 00:01:53.0            8  22ffbbb8ac8cde1   
4997     sms     25  2012-05-08 00:01:53.0            0  bcf512a3542d42c   
4998     sms      0  2012-05-08 00:01:53.0            0  05e3c3d41628a9f   
4999    data      0  2012-05-08 00:01:53.0            _  86a4dbf018f6f24   
5000   Voice      0  2012-05-08 00:01:53.0            0  a03c9837c62a5b9   

          DW_B_NUMBER COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  
0     1aa7ae01386e9fb     19e2e     19e2e  +854AcBQT2  +854AcBQT2  
1     87d87309c15a8c9     19e2e     19e2e  +854AcBQT2  +854AcBQT2  
2     93343ba6e41af95     19e2e     19e2e  +laSrk7g6q  +laSrk7g6q  
3     7d05a38f89c49d8     19e2e     19e2e  /xvRrCVKoQ  /xvRrCVKoQ  
4     56a5d5430d2fa86     19e2e     19e2e  +bO+qkSonO  +bO+qkSonO  
...               ...       ...       ...         ...         ...  
4996  8db839a0c08bd54     19e2e     19e2e  896459aace  b7d00e59e4  
4997  5a2848d6f983c31     19e2e     19e2e  7e90b35166  211015b14d  
4998  7185a95077eae9e     19e2e     19e2e  fe70e431df  c7ea989264  
4999  5a4dc9e82deb309     19e2e     OTHER  ffa6759bb2         NaN  
5000  de590e98879fa37     19e2e     19e2e  8630058aca  b47a998a3e  

[5001 rows x 10 columns]

In [132]:
Telcom3 =  pd.read_csv('Telcom_dataset.csv')
Telcom3 = pd.DataFrame(Telcom3)
Telcom3

PRODUTC  VALUE               DATETIME CELL_ON_SITE  DW_A_NUMBER_INT  \
0      Voice      0  2012-05-06 23:04:37.0            1  49fae048b547780   
1        sms      0  2012-05-06 23:05:25.0            0  7a54a6eaee19e65   
2        sms     25  2012-05-06 23:05:41.0            0  6008887b0207d82   
3      Voice     56  2012-05-06 23:05:42.0            1  7370032e20f4e96   
4      Voice     14  2012-05-06 23:06:29.0            1  966ba4656e38e62   
...      ...    ...                    ...          ...              ...   
4996     sms    100  2012-05-07 00:01:54.0            0  92e367d671f45c5   
4997   Voice      1  2012-05-07 00:01:54.0            1  83b44b18238e602   
4998   Voice      0  2012-05-07 00:01:54.0            1  49fae048b547780   
4999   Voice      0  2012-05-07 00:01:54.0            0  a1613f1a992716a   
5000   Voice      7  2012-05-07 00:01:54.0            2  1670a5f7c22f4ed   

      DW_B_NUMBER_INT COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  
0     1a49f8b0ded9de6     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  
1     93343ba6e41af95     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41  
2     7aacc9709ce0a20     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi  
3     6af4a60e74c91d3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3  
4     02aa736ee555d5c     19e2e     19e2e  051KhYDCpv  051KhYDCpv  
...               ...       ...       ...         ...         ...  
4996  38811081cf82c90     19e2e     OTHER  0c2371d9bc  1e751d08a4  
4997  6ec63d4aa5de5dd     19e2e     19e2e  5fb791cf56  65efe2bc62  
4998  1a49f8b0ded9de6     19e2e     19e2e  7773627c12  9ef9616350  
4999  bd663f07b6c8bcd     19e2e     19e2e  3f4af49d6c  969b8d0e39  
5000  3600efd094f2225     19e2e     19e2e  6f4d9b77a5  ebfc58af70  

[5001 rows x 10 columns]

In [133]:
#Updating the column names to make the uniform aand correct tyoing errors
Telcom3.columns =['PRODUCT','VALUE','DATE_TIME','CELL_ON_SITE','DW_A_NUMBER','DW_B_NUMBER','COUNTRY_A','COUNTRY_B','CELL_ID','SITE_ID']
Telcom3

PRODUCT  VALUE              DATE_TIME CELL_ON_SITE      DW_A_NUMBER  \
0      Voice      0  2012-05-06 23:04:37.0            1  49fae048b547780   
1        sms      0  2012-05-06 23:05:25.0            0  7a54a6eaee19e65   
2        sms     25  2012-05-06 23:05:41.0            0  6008887b0207d82   
3      Voice     56  2012-05-06 23:05:42.0            1  7370032e20f4e96   
4      Voice     14  2012-05-06 23:06:29.0            1  966ba4656e38e62   
...      ...    ...                    ...          ...              ...   
4996     sms    100  2012-05-07 00:01:54.0            0  92e367d671f45c5   
4997   Voice      1  2012-05-07 00:01:54.0            1  83b44b18238e602   
4998   Voice      0  2012-05-07 00:01:54.0            1  49fae048b547780   
4999   Voice      0  2012-05-07 00:01:54.0            0  a1613f1a992716a   
5000   Voice      7  2012-05-07 00:01:54.0            2  1670a5f7c22f4ed   

          DW_B_NUMBER COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  
0     1a49f8b0ded9de6     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  
1     93343ba6e41af95     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41  
2     7aacc9709ce0a20     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi  
3     6af4a60e74c91d3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3  
4     02aa736ee555d5c     19e2e     19e2e  051KhYDCpv  051KhYDCpv  
...               ...       ...       ...         ...         ...  
4996  38811081cf82c90     19e2e     OTHER  0c2371d9bc  1e751d08a4  
4997  6ec63d4aa5de5dd     19e2e     19e2e  5fb791cf56  65efe2bc62  
4998  1a49f8b0ded9de6     19e2e     19e2e  7773627c12  9ef9616350  
4999  bd663f07b6c8bcd     19e2e     19e2e  3f4af49d6c  969b8d0e39  
5000  3600efd094f2225     19e2e     19e2e  6f4d9b77a5  ebfc58af70  

[5001 rows x 10 columns]

Concatinating the 3 telcom datast tables as the have the same information

In [134]:
new = pd.concat([Telcom1, Telcom2])
new

PRODUCT  VALUE              DATE_TIME CELL_ON_SITE      DW_A_NUMBER  \
0      Voice      0  2012-05-06 23:04:37.0            1  49fae048b547780   
1        sms      0  2012-05-06 23:05:25.0            0  7a54a6eaee19e65   
2        sms     25  2012-05-06 23:05:41.0            0  6008887b0207d82   
3      Voice     56  2012-05-06 23:05:42.0            1  7370032e20f4e96   
4      Voice     14  2012-05-06 23:06:29.0            1  966ba4656e38e62   
...      ...    ...                    ...          ...              ...   
4996     sms      0  2012-05-08 00:01:53.0            8  22ffbbb8ac8cde1   
4997     sms     25  2012-05-08 00:01:53.0            0  bcf512a3542d42c   
4998     sms      0  2012-05-08 00:01:53.0            0  05e3c3d41628a9f   
4999    data      0  2012-05-08 00:01:53.0            _  86a4dbf018f6f24   
5000   Voice      0  2012-05-08 00:01:53.0            0  a03c9837c62a5b9   

          DW_B_NUMBER COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  
0     1a49f8b0ded9de6     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  
1     93343ba6e41af95     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41  
2     7aacc9709ce0a20     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi  
3     6af4a60e74c91d3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3  
4     02aa736ee555d5c     19e2e     19e2e  051KhYDCpv  051KhYDCpv  
...               ...       ...       ...         ...         ...  
4996  8db839a0c08bd54     19e2e     19e2e  896459aace  b7d00e59e4  
4997  5a2848d6f983c31     19e2e     19e2e  7e90b35166  211015b14d  
4998  7185a95077eae9e     19e2e     19e2e  fe70e431df  c7ea989264  
4999  5a4dc9e82deb309     19e2e     OTHER  ffa6759bb2         NaN  
5000  de590e98879fa37     19e2e     19e2e  8630058aca  b47a998a3e  

[10002 rows x 10 columns]

In [135]:
dftel = pd.concat([new, Telcom3])
dftel

PRODUCT  VALUE              DATE_TIME CELL_ON_SITE      DW_A_NUMBER  \
0      Voice      0  2012-05-06 23:04:37.0            1  49fae048b547780   
1        sms      0  2012-05-06 23:05:25.0            0  7a54a6eaee19e65   
2        sms     25  2012-05-06 23:05:41.0            0  6008887b0207d82   
3      Voice     56  2012-05-06 23:05:42.0            1  7370032e20f4e96   
4      Voice     14  2012-05-06 23:06:29.0            1  966ba4656e38e62   
...      ...    ...                    ...          ...              ...   
4996     sms    100  2012-05-07 00:01:54.0            0  92e367d671f45c5   
4997   Voice      1  2012-05-07 00:01:54.0            1  83b44b18238e602   
4998   Voice      0  2012-05-07 00:01:54.0            1  49fae048b547780   
4999   Voice      0  2012-05-07 00:01:54.0            0  a1613f1a992716a   
5000   Voice      7  2012-05-07 00:01:54.0            2  1670a5f7c22f4ed   

          DW_B_NUMBER COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  
0     1a49f8b0ded9de6     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  
1     93343ba6e41af95     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41  
2     7aacc9709ce0a20     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi  
3     6af4a60e74c91d3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3  
4     02aa736ee555d5c     19e2e     19e2e  051KhYDCpv  051KhYDCpv  
...               ...       ...       ...         ...         ...  
4996  38811081cf82c90     19e2e     OTHER  0c2371d9bc  1e751d08a4  
4997  6ec63d4aa5de5dd     19e2e     19e2e  5fb791cf56  65efe2bc62  
4998  1a49f8b0ded9de6     19e2e     19e2e  7773627c12  9ef9616350  
4999  bd663f07b6c8bcd     19e2e     19e2e  3f4af49d6c  969b8d0e39  
5000  3600efd094f2225     19e2e     19e2e  6f4d9b77a5  ebfc58af70  

[15003 rows x 10 columns]

In [136]:
alldata = dftel.merge(cellgeo, left_index= True, right_index = True)
alldata

PRODUCT  VALUE              DATE_TIME CELL_ON_SITE      DW_A_NUMBER  \
0      Voice      0  2012-05-06 23:04:37.0            1  49fae048b547780   
0        sms      0  2012-05-07 23:02:06.0            2  32255675c3b6fe3   
0      Voice      0  2012-05-06 23:04:37.0            1  49fae048b547780   
1        sms      0  2012-05-06 23:05:25.0            0  7a54a6eaee19e65   
1      Voice      0  2012-05-07 23:03:44.0            1  ff16d6ac74c1c64   
...      ...    ...                    ...          ...              ...   
3972   Voice     34  2012-05-08 00:01:25.0            1  f90739859eb4852   
3972   Voice     29  2012-05-07 00:01:26.0            2  d63f6042e75d51b   
3973   Voice     40  2012-05-07 00:01:26.0            5  6aa773f400a4b1b   
3973   Voice      0  2012-05-08 00:01:25.0            0  474bdfd8355e807   
3973   Voice     40  2012-05-07 00:01:26.0            5  6aa773f400a4b1b   

          DW_B_NUMBER COUNTRY_A COUNTRY_B   CELL_ID_x     SITE_ID  ...  \
0     1a49f8b0ded9de6     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  ...   
0     1aa7ae01386e9fb     19e2e     19e2e  +854AcBQT2  +854AcBQT2  ...   
0     1a49f8b0ded9de6     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  ...   
1     93343ba6e41af95     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41  ...   
1     87d87309c15a8c9     19e2e     19e2e  +854AcBQT2  +854AcBQT2  ...   
...               ...       ...       ...         ...         ...  ...   
3972  1a465a6dbbcd519     19e2e     19e2e  b66b4efcdc  447989361c  ...   
3972  35b093b6a4549da     19e2e     19e2e  706d8914d0  22389546ec  ...   
3973  88dd51e4d6885f0     19e2e     19e2e  3424273a07  ae6cdb80c1  ...   
3973  4571828cc5decad     19e2e     19e2e  bc1d885e30  a7bd4c027b  ...   
3973  88dd51e4d6885f0     19e2e     19e2e  3424273a07  ae6cdb80c1  ...   

          STATUS LOCALISATION    DECOUPZONE             ZONENAME LONGITUDE  \
0     In Service      ABIDJAN  "Abidjan_EST             AGBANNEW -4.000667   
0     In Service      ABIDJAN  "Abidjan_EST             AGBANNEW -4.000667   
0     In Service      ABIDJAN  "Abidjan_EST             AGBANNEW -4.000667   
1     In Service      ABIDJAN  "Abidjan_EST             AGBANNEW -4.000667   
1     In Service      ABIDJAN  "Abidjan_EST             AGBANNEW -4.000667   
...          ...          ...           ...                  ...       ...   
3972  In Service    INTERIEUR    Grand-NORD  ZONE A - Grand-NORD -6.065450   
3972  In Service    INTERIEUR    Grand-NORD  ZONE A - Grand-NORD -6.065450   
3973  In Service    INTERIEUR    Grand-NORD  ZONE A - Grand-NORD -6.065450   
3973  In Service    INTERIEUR    Grand-NORD  ZONE A - Grand-NORD -6.065450   
3973  In Service    INTERIEUR    Grand-NORD  ZONE A - Grand-NORD -6.065450   

      LATITUDE    REGION          AREA   CELL_ID_y   SITE_CODE  
0     5.368194    LAGUNE           NaN  fbb2e75af3  0bb492ef17  
0     5.368194    LAGUNE           NaN  fbb2e75af3  0bb492ef17  
0     5.368194    LAGUNE           NaN  fbb2e75af3  0bb492ef17  
1     5.368194    LAGUNE           NaN  1502501d78  0bb492ef17  
1     5.368194    LAGUNE           NaN  1502501d78  0bb492ef17  
...        ...       ...           ...         ...         ...  
3972  7.444592  MARAHOUE  CENTRE NORTH  d4a6a0b645  61bee9c70c  
3972  7.444592  MARAHOUE  CENTRE NORTH  d4a6a0b645  61bee9c70c  
3973  7.444592  MARAHOUE  CENTRE NORTH  f9f3e42589  61bee9c70c  
3973  7.444592  MARAHOUE  CENTRE NORTH  f9f3e42589  61bee9c70c  
3973  7.444592  MARAHOUE  CENTRE NORTH  f9f3e42589  61bee9c70c  

[11922 rows x 22 columns]

In [137]:
alldata.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11922 entries, 0 to 3973
Data columns (total 22 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   PRODUCT       11922 non-null  object 
 1   VALUE         11922 non-null  int64  
 2   DATE_TIME     11922 non-null  object 
 3   CELL_ON_SITE  11922 non-null  object 
 4   DW_A_NUMBER   11922 non-null  object 
 5   DW_B_NUMBER   11922 non-null  object 
 6   COUNTRY_A     11922 non-null  object 
 7   COUNTRY_B     11922 non-null  object 
 8   CELL_ID_x     11922 non-null  object 
 9   SITE_ID       10295 non-null  object 
 10  Unnamed: 0    11922 non-null  int64  
 11  VILLES        11922 non-null  object 
 12  STATUS        11721 non-null  object 
 13  LOCALISATION  11922 non-null  object 
 14  DECOUPZONE    11922 non-null  object 
 15  ZONENAME      11904 non-null  object 
 16  LONGITUDE     11922 non-null  float64
 17  LATITUDE      11922 non-null  float64
 18  REGION        11922 non-nul

DATA UNDERSTANDING

In [138]:
#information on the cell geographics table
cellgeo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3974 entries, 0 to 3973
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Unnamed: 0    3974 non-null   int64  
 1   VILLES        3974 non-null   object 
 2   STATUS        3907 non-null   object 
 3   LOCALISATION  3974 non-null   object 
 4   DECOUPZONE    3974 non-null   object 
 5   ZONENAME      3968 non-null   object 
 6   LONGITUDE     3974 non-null   float64
 7   LATITUDE      3974 non-null   float64
 8   REGION        3974 non-null   object 
 9   AREA          3951 non-null   object 
 10  CELL_ID       3974 non-null   object 
 11  SITE_CODE     3974 non-null   object 
dtypes: float64(2), int64(1), object(9)
memory usage: 372.7+ KB


the table above has 3,974 rows in 12 columns.
Status column has 67 missing entries
zone name has 6 missing value
area column has 23 missing entries


In [139]:
#Information on the telcom datasets
dftel.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 15003 entries, 0 to 5000
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   PRODUCT       15003 non-null  object
 1   VALUE         15003 non-null  int64 
 2   DATE_TIME     15003 non-null  object
 3   CELL_ON_SITE  15003 non-null  object
 4   DW_A_NUMBER   15003 non-null  object
 5   DW_B_NUMBER   15003 non-null  object
 6   COUNTRY_A     15003 non-null  object
 7   COUNTRY_B     15003 non-null  object
 8   CELL_ID       15003 non-null  object
 9   SITE_ID       12895 non-null  object
dtypes: int64(1), object(9)
memory usage: 1.5+ MB


the telcom dataset has 15,003 rows and 9 columns
the site_id column has 2,108 missing vavues


In [140]:
#drop non required colomns
alldata = alldata.drop(columns=['CELL_ON_SITE'])
alldata = alldata.drop(columns=['COUNTRY_A'])
alldata = alldata.drop(columns=['COUNTRY_B'])
alldata = alldata.drop(columns = ['Unnamed: 0'])

DATA CLEANING

dropping entriels with null values

In [141]:
dftel.dropna(inplace = True)

In [142]:
cellgeo.dropna(inplace = True)


In [143]:
alldata.dropna(inplace = True)

In [144]:
#set the date/time format to ensure all date/time entries are in the right format
dftel['DATE_TIME'] = pd.to_datetime(dftel['DATE_TIME'])
dftel.head(5)


PRODUCT  VALUE           DATE_TIME CELL_ON_SITE      DW_A_NUMBER  \
0   Voice      0 2012-05-06 23:04:37            1  49fae048b547780   
1     sms      0 2012-05-06 23:05:25            0  7a54a6eaee19e65   
2     sms     25 2012-05-06 23:05:41            0  6008887b0207d82   
3   Voice     56 2012-05-06 23:05:42            1  7370032e20f4e96   
4   Voice     14 2012-05-06 23:06:29            1  966ba4656e38e62   

       DW_B_NUMBER COUNTRY_A COUNTRY_B     CELL_ID     SITE_ID  
0  1a49f8b0ded9de6     19e2e     19e2e  /TJNe+Mmtu  /TJNe+Mmtu  
1  93343ba6e41af95     19e2e     19e2e  /+cKZKIp41  /+cKZKIp41  
2  7aacc9709ce0a20     19e2e     19e2e  /u0FSD+ahi  /u0FSD+ahi  
3  6af4a60e74c91d3     19e2e     19e2e  +cNeJzsTp3  +cNeJzsTp3  
4  02aa736ee555d5c     19e2e     19e2e  051KhYDCpv  051KhYDCpv

In [145]:
#set the date/time format to ensure all date/time entries are in the right format
alldata['DATE_TIME'] = pd.to_datetime(alldata['DATE_TIME'])
alldata.head(3)

PRODUCT  VALUE           DATE_TIME      DW_A_NUMBER      DW_B_NUMBER  \
23    data      0 2012-05-06 23:11:27  795c2a90bd8d442  5a4dc9e82deb309   
23     sms      0 2012-05-07 23:09:50  8da7870ee027a39  43856ca73fbadf3   
23    data      0 2012-05-06 23:11:27  795c2a90bd8d442  5a4dc9e82deb309   

     CELL_ID_x     SITE_ID VILLES      STATUS LOCALISATION    DECOUPZONE  \
23  2yuhUrSXQu  2yuhUrSXQu  ABOBO  In Service      ABIDJAN  Abidjan_NORD   
23  +854AcBQT2  +854AcBQT2  ABOBO  In Service      ABIDJAN  Abidjan_NORD   
23  2yuhUrSXQu  2yuhUrSXQu  ABOBO  In Service      ABIDJAN  Abidjan_NORD   

                 ZONENAME  LONGITUDE  LATITUDE  REGION           AREA  \
23  ZONE E - Abidjan_NORD  -4.014139    5.4335  LAGUNE  ABIDJAN NORTH   
23  ZONE E - Abidjan_NORD  -4.014139    5.4335  LAGUNE  ABIDJAN NORTH   
23  ZONE E - Abidjan_NORD  -4.014139    5.4335  LAGUNE  ABIDJAN NORTH   

     CELL_ID_y   SITE_CODE  
23  f740406877  0f34c36ac5  
23  f740406877  0f34c36ac5  
23  f740406877  0f34c36ac5

In [146]:
#check for duplicates
print(cellgeo.duplicated())

23      False
24      False
25      False
26      False
27      False
        ...  
3969    False
3970    False
3971    False
3972    False
3973    False
Length: 3880, dtype: bool


In [147]:
#Check for duplicates
dftel.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
4996     True
4997     True
4998     True
4999     True
5000     True
Length: 12895, dtype: bool

DATA ANALYSIS

Grouping

In [148]:
#Groupint the cell geography table by city
villes = alldata.groupby('VILLES')
villes.first()

PRODUCT  VALUE           DATE_TIME      DW_A_NUMBER  \
VILLES                                                              
ABENGOUROU        sms      0 2012-05-06 23:14:40  d6dec69142f5f5a   
ABOBO            data      0 2012-05-06 23:11:27  795c2a90bd8d442   
ABOISSO         Voice      0 2012-05-06 23:28:31  8289c0a4598fcf7   
ABOISSO COMOE     sms      0 2012-05-06 23:28:12  5bc1392173563bb   
ABONGOUA          sms     25 2012-05-06 23:29:53  6d99688d1ff9e7a   
...               ...    ...                 ...              ...   
ZATTA             sms      0 2012-05-07 00:01:26  74f4e2d12e8d8e6   
ZIKISSO           sms      0 2012-05-07 00:01:26  8ab8b9ed6989f61   
ZON-HOUNIEN       sms      0 2012-05-07 00:01:26  7e9c1c707d4310f   
ZOUKOUGBEU      Voice      0 2012-05-07 00:01:26  189da05b0b88918   
ZUENOULA        Voice     26 2012-05-07 00:01:10  48117094e870054   

                   DW_B_NUMBER   CELL_ID_x     SITE_ID      STATUS  \
VILLES                                                               
ABENGOUROU     679f88c531065ba  0eNnL1pokg  0eNnL1pokg  In Service   
ABOBO          5a4dc9e82deb309  2yuhUrSXQu  2yuhUrSXQu  In Service   
ABOISSO        36444e531eb6d38  0eA3S0GqHt  0eA3S0GqHt  In Service   
ABOISSO COMOE  158fac5b0923134  6/1Gti3zBg  6/1Gti3zBg  In Service   
ABONGOUA       113e0b82148aa73  3IoIGy1+o8  3IoIGy1+o8  In Service   
...                        ...         ...         ...         ...   
ZATTA          d77881f35d3f676  fa9f48bbfd  6560f866a3  In Service   
ZIKISSO        e09e736e46c5c3d  253f3e83ab  f11e961893  In Service   
ZON-HOUNIEN    238a3effcf6a3dc  c3a6d994b3  d29252461f  In Service   
ZOUKOUGBEU     8a70e8bb8834081  f8f987f764  651c6d67d4  In Service   
ZUENOULA       50592ddc5ea2b9c  d1889b433b  e516cb8121  In Service   

              LOCALISATION    DECOUPZONE               ZONENAME  LONGITUDE  \
VILLES                                                                       
ABENGOUROU       INTERIEUR     Grand-EST     ZONE C - Grand-EST  -3.514833   
ABOBO              ABIDJAN  Abidjan_NORD  ZONE E - Abidjan_NORD  -4.014139   
ABOISSO          INTERIEUR     Grand-EST     ZONE C - Grand-EST  -0.000000   
ABOISSO COMOE    INTERIEUR     Grand-EST     ZONE C - Grand-EST  -3.396861   
ABONGOUA         INTERIEUR     Grand-EST     ZONE C - Grand-EST  -3.548722   
...                    ...           ...                    ...        ...   
ZATTA            INTERIEUR    Grand-NORD    ZONE B - Grand-NORD  -5.415614   
ZIKISSO          INTERIEUR     Grand-SUD     ZONE B - Grand-SUD  -5.698833   
ZON-HOUNIEN      INTERIEUR   Grand-OUEST   ZONE D - Grand-OUEST  -8.205167   
ZOUKOUGBEU       INTERIEUR   Grand-OUEST   ZONE D - Grand-OUEST  -6.865694   
ZUENOULA         INTERIEUR    Grand-NORD    ZONE A - Grand-NORD  -6.138956   

               LATITUDE          REGION           AREA   CELL_ID_y   SITE_CODE  
VILLES                                                                          
ABENGOUROU     6.722333     MOYEN COMOE           EAST  8cdfc1bc65  6e87d6cb59  
ABOBO          5.433500          LAGUNE  ABIDJAN NORTH  f740406877  0f34c36ac5  
ABOISSO        0.000000       SUD COMOE  ABIDJAN SOUTH  b4d8c16d9e  38c5c1de07  
ABOISSO COMOE  5.734500          LAGUNE  ABIDJAN NORTH  ff4372eafb  640427c4c9  
ABONGOUA       6.247083     MOYEN COMOE           EAST  d1a1270467  a8d8462dda  
...                 ...             ...            ...         ...         ...  
ZATTA          6.905422            LACS   CENTRE NORTH  1d41c565fc  c644d1ce4c  
ZIKISSO        6.057278     SUD BANDAMA     SOUTH WEST  733df73327  211015b14d  
ZON-HOUNIEN    6.920500    18 MONTAGNES     NORTH WEST  a3e5534191  32b8357a82  
ZOUKOUGBEU     6.761306  HAUT SASSANDRA     NORTH WEST  317ba2e3e8  443ac3fc19  
ZUENOULA       7.617553        MARAHOUE   CENTRE NORTH  8cde542067  43adae9a57  

[495 rows x 17 columns]

In [149]:
#use count() function to determine the number of cell activities recorded per city
villescount = villes.count()

In [150]:
# sort the grouping count in descending to see cities that had the most cell ussage in the 3 days.
dv = villescount.sort_values('CELL_ID_y', ascending = False)
#use head() to get the top 10 cities
dv.head(10)

PRODUCT  VALUE  DATE_TIME  DW_A_NUMBER  DW_B_NUMBER  CELL_ID_x  \
VILLES                                                                         
COCODY            917    917        917          917          917        917   
YOPOUGON          759    759        759          759          759        759   
ABOBO             696    696        696          696          696        696   
MARCORY           359    359        359          359          359        359   
TREICHVILLE       337    337        337          337          337        337   
ADJAME            286    286        286          286          286        286   
PLATEAU           277    277        277          277          277        277   
YAMOUSSOUKRO      267    267        267          267          267        267   
KOUMASSI          233    233        233          233          233        233   
PORT-BOUET        163    163        163          163          163        163   

              SITE_ID  STATUS  LOCALISATION  DECOUPZONE  ZONENAME  LONGITUDE  \
VILLES                                                                         
COCODY            917     917           917         917       917        917   
YOPOUGON          759     759           759         759       759        759   
ABOBO             696     696           696         696       696        696   
MARCORY           359     359           359         359       359        359   
TREICHVILLE       337     337           337         337       337        337   
ADJAME            286     286           286         286       286        286   
PLATEAU           277     277           277         277       277        277   
YAMOUSSOUKRO      267     267           267         267       267        267   
KOUMASSI          233     233           233         233       233        233   
PORT-BOUET        163     163           163         163       163        163   

              LATITUDE  REGION  AREA  CELL_ID_y  SITE_CODE  
VILLES                                                      
COCODY             917     917   917        917        917  
YOPOUGON           759     759   759        759        759  
ABOBO              696     696   696        696        696  
MARCORY            359     359   359        359        359  
TREICHVILLE        337     337   337        337        337  
ADJAME             286     286   286        286        286  
PLATEAU            277     277   277        277        277  
YAMOUSSOUKRO       267     267   267        267        267  
KOUMASSI           233     233   233        233        233  
PORT-BOUET         163     163   163        163        163

In [151]:
#from the above table grouped by citites/villes, group by hour using .resample to see what times the users were using in each city.
vt = villes.resample('H', on = 'DATE_TIME')
vtcount = vt.count()
vs = vtcount.sort_values('PRODUCT', ascending = False)
vs.head(20)

PRODUCT  VALUE  DATE_TIME  DW_A_NUMBER  \
VILLES       DATE_TIME                                                     
COCODY       2012-05-07 00:00:00      466    466        466          466   
YOPOUGON     2012-05-07 00:00:00      456    456        456          456   
ABOBO        2012-05-06 23:00:00      322    322        322          322   
COCODY       2012-05-08 00:00:00      259    259        259          259   
TREICHVILLE  2012-05-07 00:00:00      226    226        226          226   
YOPOUGON     2012-05-08 00:00:00      216    216        216          216   
MARCORY      2012-05-07 00:00:00      180    180        180          180   
YAMOUSSOUKRO 2012-05-07 00:00:00      178    178        178          178   
ABOBO        2012-05-07 23:00:00      161    161        161          161   
PLATEAU      2012-05-07 00:00:00      160    160        160          160   
KOUMASSI     2012-05-07 00:00:00      152    152        152          152   
ABOBO        2012-05-07 00:00:00      140    140        140          140   
COCODY       2012-05-06 23:00:00      128    128        128          128   
ADJAME       2012-05-07 00:00:00      114    114        114          114   
TREICHVILLE  2012-05-08 00:00:00      111    111        111          111   
GAGNOA       2012-05-07 00:00:00      102    102        102          102   
MARCORY      2012-05-08 00:00:00      100    100        100          100   
DALOA        2012-05-07 00:00:00       90     90         90           90   
SAN-PEDRO    2012-05-07 00:00:00       88     88         88           88   
YAMOUSSOUKRO 2012-05-08 00:00:00       83     83         83           83   

                                  DW_B_NUMBER  CELL_ID_x  SITE_ID  VILLES  \
VILLES       DATE_TIME                                                      
COCODY       2012-05-07 00:00:00          466        466      466     466   
YOPOUGON     2012-05-07 00:00:00          456        456      456     456   
ABOBO        2012-05-06 23:00:00          322        322      322     322   
COCODY       2012-05-08 00:00:00          259        259      259     259   
TREICHVILLE  2012-05-07 00:00:00          226        226      226     226   
YOPOUGON     2012-05-08 00:00:00          216        216      216     216   
MARCORY      2012-05-07 00:00:00          180        180      180     180   
YAMOUSSOUKRO 2012-05-07 00:00:00          178        178      178     178   
ABOBO        2012-05-07 23:00:00          161        161      161     161   
PLATEAU      2012-05-07 00:00:00          160        160      160     160   
KOUMASSI     2012-05-07 00:00:00          152        152      152     152   
ABOBO        2012-05-07 00:00:00          140        140      140     140   
COCODY       2012-05-06 23:00:00          128        128      128     128   
ADJAME       2012-05-07 00:00:00          114        114      114     114   
TREICHVILLE  2012-05-08 00:00:00          111        111      111     111   
GAGNOA       2012-05-07 00:00:00          102        102      102     102   
MARCORY      2012-05-08 00:00:00          100        100      100     100   
DALOA        2012-05-07 00:00:00           90         90       90      90   
SAN-PEDRO    2012-05-07 00:00:00           88         88       88      88   
YAMOUSSOUKRO 2012-05-08 00:00:00           83         83       83      83   

                                  STATUS  LOCALISATION  DECOUPZONE  ZONENAME  \
VILLES       DATE_TIME                                                         
COCODY       2012-05-07 00:00:00     466           466         466       466   
YOPOUGON     2012-05-07 00:00:00     456           456         456       456   
ABOBO        2012-05-06 23:00:00     322           322         322       322   
COCODY       2012-05-08 00:00:00     259           259         259       259   
TREICHVILLE  2012-05-07 00:00:00     226           226         226       226   
YOPOUGON     2012-05-08 00:00:00     216           216         216       216   
MARCORY      2012-05-07 00

In [152]:
vs.tail(20)

PRODUCT  VALUE  DATE_TIME  DW_A_NUMBER  \
VILLES   DATE_TIME                                                     
FOUMBOLO 2012-05-07 10:00:00        0      0          0            0   
         2012-05-07 08:00:00        0      0          0            0   
FLAKIE   2012-05-07 15:00:00        0      0          0            0   
FOUMBOLO 2012-05-07 07:00:00        0      0          0            0   
FLAKIE   2012-05-07 16:00:00        0      0          0            0   
         2012-05-07 17:00:00        0      0          0            0   
         2012-05-07 18:00:00        0      0          0            0   
         2012-05-07 19:00:00        0      0          0            0   
         2012-05-07 20:00:00        0      0          0            0   
         2012-05-07 21:00:00        0      0          0            0   
         2012-05-07 22:00:00        0      0          0            0   
         2012-05-07 23:00:00        0      0          0            0   
ADJAMENE 2012-05-07 08:00:00        0      0          0            0   
FOUMBOLO 2012-05-07 01:00:00        0      0          0            0   
         2012-05-07 02:00:00        0      0          0            0   
         2012-05-07 03:00:00        0      0          0            0   
         2012-05-07 04:00:00        0      0          0            0   
         2012-05-07 05:00:00        0      0          0            0   
         2012-05-07 06:00:00        0      0          0            0   
KODJINAN 2012-05-07 13:00:00        0      0          0            0   

                              DW_B_NUMBER  CELL_ID_x  SITE_ID  VILLES  STATUS  \
VILLES   DATE_TIME                                                              
FOUMBOLO 2012-05-07 10:00:00            0          0        0       0       0   
         2012-05-07 08:00:00            0          0        0       0       0   
FLAKIE   2012-05-07 15:00:00            0          0        0       0       0   
FOUMBOLO 2012-05-07 07:00:00            0          0        0       0       0   
FLAKIE   2012-05-07 16:00:00            0          0        0       0       0   
         2012-05-07 17:00:00            0          0        0       0       0   
         2012-05-07 18:00:00            0          0        0       0       0   
         2012-05-07 19:00:00            0          0        0       0       0   
         2012-05-07 20:00:00            0          0        0       0       0   
         2012-05-07 21:00:00            0          0        0       0       0   
         2012-05-07 22:00:00            0          0        0       0       0   
         2012-05-07 23:00:00            0          0        0       0       0   
ADJAMENE 2012-05-07 08:00:00            0          0        0       0       0   
FOUMBOLO 2012-05-07 01:00:00            0          0        0       0       0   
         2012-05-07 02:00:00            0          0        0       0       0   
         2012-05-07 03:00:00            0          0        0       0       0   
         2012-05-07 04:00:00            0          0        0       0       0   
         2012-05-07 05:00:00            0          0        0       0       0   
         2012-05-07 06:00:00            0          0        0       0       0   
KODJINAN 2012-05-07 13:00:00            0          0        0       0       0   

                              LOCALISATION  DECOUPZONE  ZONENAME  LONGITUDE  \
VILLES   DATE_TIME                                                            
FOUMBOLO 2012-05-07 10:00:00             0           0         0          0   
         2012-05-07 08:00:00             0           0         0          0   
FLAKIE   2012-05-07 15:00:00             0           0         0          0   
FOUMBOLO 2012-05-07 07:00:00             0           0         0          0   
FLAKIE   2012-05-07 16:00:00             0           0         0          0   
         2012-05-07 17:00:00             0           0         0          0   
         2012-05-07 18:00:00  

In [153]:
# describing the group count to see the statistics of all data collected in the three days.
dv.describe()


PRODUCT       VALUE   DATE_TIME  DW_A_NUMBER  DW_B_NUMBER  \
count  495.000000  495.000000  495.000000   495.000000   495.000000   
mean    20.280808   20.280808   20.280808    20.280808    20.280808   
std     70.684707   70.684707   70.684707    70.684707    70.684707   
min      1.000000    1.000000    1.000000     1.000000     1.000000   
25%      7.000000    7.000000    7.000000     7.000000     7.000000   
50%      9.000000    9.000000    9.000000     9.000000     9.000000   
75%      9.000000    9.000000    9.000000     9.000000     9.000000   
max    917.000000  917.000000  917.000000   917.000000   917.000000   

        CELL_ID_x     SITE_ID      STATUS  LOCALISATION  DECOUPZONE  \
count  495.000000  495.000000  495.000000    495.000000  495.000000   
mean    20.280808   20.280808   20.280808     20.280808   20.280808   
std     70.684707   70.684707   70.684707     70.684707   70.684707   
min      1.000000    1.000000    1.000000      1.000000    1.000000   
25%      7.000000    7.000000    7.000000      7.000000    7.000000   
50%      9.000000    9.000000    9.000000      9.000000    9.000000   
75%      9.000000    9.000000    9.000000      9.000000    9.000000   
max    917.000000  917.000000  917.000000    917.000000  917.000000   

         ZONENAME   LONGITUDE    LATITUDE      REGION        AREA   CELL_ID_y  \
count  495.000000  495.000000  495.000000  495.000000  495.000000  495.000000   
mean    20.280808   20.280808   20.280808   20.280808   20.280808   20.280808   
std     70.684707   70.684707   70.684707   70.684707   70.684707   70.684707   
min      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   
25%      7.000000    7.000000    7.000000    7.000000    7.000000    7.000000   
50%      9.000000    9.000000    9.000000    9.000000    9.000000    9.000000   
75%      9.000000    9.000000    9.000000    9.000000    9.000000    9.000000   
max    917.000000  917.000000  917.000000  917.000000  917.000000  917.000000   

        SITE_CODE  
count  495.000000  
mean    20.280808  
std     70.684707  
min      1.000000  
25%      7.000000  
50%      9.000000  
75%      9.000000  
max    917.000000

In [154]:
#use tail() to filter the cities that recorded the least number of cell phone activity.
dv.tail(5)

PRODUCT  VALUE  DATE_TIME  DW_A_NUMBER  DW_B_NUMBER  \
VILLES                                                                   
DANANON                  2      2          2            2            2   
GABIADJI                 2      2          2            2            2   
TORO-KINKENE             1      1          1            1            1   
ZARANOU                  1      1          1            1            1   
COLLINEKASSEREGUE        1      1          1            1            1   

                   CELL_ID_x  SITE_ID  STATUS  LOCALISATION  DECOUPZONE  \
VILLES                                                                    
DANANON                    2        2       2             2           2   
GABIADJI                   2        2       2             2           2   
TORO-KINKENE               1        1       1             1           1   
ZARANOU                    1        1       1             1           1   
COLLINEKASSEREGUE          1        1       1             1           1   

                   ZONENAME  LONGITUDE  LATITUDE  REGION  AREA  CELL_ID_y  \
VILLES                                                                      
DANANON                   2          2         2       2     2          2   
GABIADJI                  2          2         2       2     2          2   
TORO-KINKENE              1          1         1       1     1          1   
ZARANOU                   1          1         1       1     1          1   
COLLINEKASSEREGUE         1          1         1       1     1          1   

                   SITE_CODE  
VILLES                        
DANANON                    2  
GABIADJI                   2  
TORO-KINKENE               1  
ZARANOU                    1  
COLLINEKASSEREGUE          1

In [155]:
#group the telcom data by products to see the most used product between voice,sms and data
tel = alldata.groupby('PRODUCT')
tel.first()

VALUE           DATE_TIME      DW_A_NUMBER      DW_B_NUMBER  \
PRODUCT                                                                
Voice        0 2012-05-06 23:11:27  1a2b54695968d46  55ad144fa74fddc   
data         0 2012-05-06 23:11:27  795c2a90bd8d442  5a4dc9e82deb309   
sms          0 2012-05-07 23:09:50  8da7870ee027a39  43856ca73fbadf3   

          CELL_ID_x     SITE_ID VILLES      STATUS LOCALISATION    DECOUPZONE  \
PRODUCT                                                                         
Voice    3mQAiRAfo7  3mQAiRAfo7  ABOBO  In Service      ABIDJAN  Abidjan_NORD   
data     2yuhUrSXQu  2yuhUrSXQu  ABOBO  In Service      ABIDJAN  Abidjan_NORD   
sms      +854AcBQT2  +854AcBQT2  ABOBO  In Service      ABIDJAN  Abidjan_NORD   

                      ZONENAME  LONGITUDE  LATITUDE  REGION           AREA  \
PRODUCT                                                                      
Voice    ZONE E - Abidjan_NORD  -4.014139    5.4335  LAGUNE  ABIDJAN NORTH   
data     ZONE E - Abidjan_NORD  -4.014139    5.4335  LAGUNE  ABIDJAN NORTH   
sms      ZONE E - Abidjan_NORD  -4.014139    5.4335  LAGUNE  ABIDJAN NORTH   

          CELL_ID_y   SITE_CODE  
PRODUCT                          
Voice    959e54c8f7  0f34c36ac5  
data     f740406877  0f34c36ac5  
sms      f740406877  0f34c36ac5

In [156]:
#use count() to see the number of times each product was usedin the three days
telcount = tel.count()
telcount

VALUE  DATE_TIME  DW_A_NUMBER  DW_B_NUMBER  CELL_ID_x  SITE_ID  \
PRODUCT                                                                   
Voice     4821       4821         4821         4821       4821     4821   
data       264        264          264          264        264      264   
sms       4954       4954         4954         4954       4954     4954   

         VILLES  STATUS  LOCALISATION  DECOUPZONE  ZONENAME  LONGITUDE  \
PRODUCT                                                                  
Voice      4821    4821          4821        4821      4821       4821   
data        264     264           264         264       264        264   
sms        4954    4954          4954        4954      4954       4954   

         LATITUDE  REGION  AREA  CELL_ID_y  SITE_CODE  
PRODUCT                                                
Voice        4821    4821  4821       4821       4821  
data          264     264   264        264        264  
sms          4954    4954  4954       4954       4954

In [157]:
#obtaining the total bill value from each product offered by MTN
tel2 = alldata.groupby('PRODUCT').sum('VALUE')
print(tel2)

          VALUE     LONGITUDE      LATITUDE
PRODUCT                                    
Voice    247243 -23438.048314  29906.272903
data       1809  -1091.235801   1524.649473
sms       31883 -24090.482407  30988.657425


In [158]:
#grouping by those in ABIDJAN and those that are not in Abidjan
local = alldata.groupby('LOCALISATION')
local.count()

PRODUCT  VALUE  DATE_TIME  DW_A_NUMBER  DW_B_NUMBER  CELL_ID_x  \
LOCALISATION                                                                   
ABIDJAN          4339   4339       4339         4339         4339       4339   
INTERIEUR        5700   5700       5700         5700         5700       5700   

              SITE_ID  VILLES  STATUS  DECOUPZONE  ZONENAME  LONGITUDE  \
LOCALISATION                                                             
ABIDJAN          4339    4339    4339        4339      4339       4339   
INTERIEUR        5700    5700    5700        5700      5700       5700   

              LATITUDE  REGION  AREA  CELL_ID_y  SITE_CODE  
LOCALISATION                                                
ABIDJAN           4339    4339  4339       4339       4339  
INTERIEUR         5700    5700  5700       5700       5700

In [159]:
#grouping by siteid
site = alldata.groupby('SITE_ID')

In [160]:
#use count() to see how many times each site id was used in the three days.
# ordered in descending order
sitecount = site.count()
siteid = sitecount.sort_values('VALUE',ascending = False)
siteid

PRODUCT  VALUE  DATE_TIME  DW_A_NUMBER  DW_B_NUMBER  CELL_ID_x  \
SITE_ID                                                                      
b47a998a3e       98     98         98           98           98         98   
a7bd4c027b       70     70         70           70           70         70   
5c565fbfb1       63     63         63           63           63         63   
7f76d66503       62     62         62           62           62         62   
585a3bc66c       54     54         54           54           54         54   
...             ...    ...        ...          ...          ...        ...   
MW5J8MT3Zs        1      1          1            1            1          1   
c9cb7e8be0        1      1          1            1            1          1   
e1629ca4b7        1      1          1            1            1          1   
9f9bb90c1f        1      1          1            1            1          1   
KE4VqUMLIn        1      1          1            1            1          1   

            VILLES  STATUS  LOCALISATION  DECOUPZONE  ZONENAME  LONGITUDE  \
SITE_ID                                                                     
b47a998a3e      98      98            98          98        98         98   
a7bd4c027b      70      70            70          70        70         70   
5c565fbfb1      63      63            63          63        63         63   
7f76d66503      62      62            62          62        62         62   
585a3bc66c      54      54            54          54        54         54   
...            ...     ...           ...         ...       ...        ...   
MW5J8MT3Zs       1       1             1           1         1          1   
c9cb7e8be0       1       1             1           1         1          1   
e1629ca4b7       1       1             1           1         1          1   
9f9bb90c1f       1       1             1           1         1          1   
KE4VqUMLIn       1       1             1           1         1          1   

            LATITUDE  REGION  AREA  CELL_ID_y  SITE_CODE  
SITE_ID                                                   
b47a998a3e        98      98    98         98         98  
a7bd4c027b        70      70    70         70         70  
5c565fbfb1        63      63    63         63         63  
7f76d66503        62      62    62         62         62  
585a3bc66c        54      54    54         54         54  
...              ...     ...   ...        ...        ...  
MW5J8MT3Zs         1       1     1          1          1  
c9cb7e8be0         1       1     1          1          1  
e1629ca4b7         1       1     1          1          1  
9f9bb90c1f         1       1     1          1          1  
KE4VqUMLIn         1       1     1          1          1  

[1230 rows x 17 columns]

In [161]:
#the top 10 most used sites in th three days
siteid.head(10)

PRODUCT  VALUE  DATE_TIME  DW_A_NUMBER  DW_B_NUMBER  CELL_ID_x  \
SITE_ID                                                                      
b47a998a3e       98     98         98           98           98         98   
a7bd4c027b       70     70         70           70           70         70   
5c565fbfb1       63     63         63           63           63         63   
7f76d66503       62     62         62           62           62         62   
585a3bc66c       54     54         54           54           54         54   
928a59b635       52     52         52           52           52         52   
c07a544c63       52     52         52           52           52         52   
8fd6f8d1c5       50     50         50           50           50         50   
bc58a52936       50     50         50           50           50         50   
91e86be9a6       50     50         50           50           50         50   

            VILLES  STATUS  LOCALISATION  DECOUPZONE  ZONENAME  LONGITUDE  \
SITE_ID                                                                     
b47a998a3e      98      98            98          98        98         98   
a7bd4c027b      70      70            70          70        70         70   
5c565fbfb1      63      63            63          63        63         63   
7f76d66503      62      62            62          62        62         62   
585a3bc66c      54      54            54          54        54         54   
928a59b635      52      52            52          52        52         52   
c07a544c63      52      52            52          52        52         52   
8fd6f8d1c5      50      50            50          50        50         50   
bc58a52936      50      50            50          50        50         50   
91e86be9a6      50      50            50          50        50         50   

            LATITUDE  REGION  AREA  CELL_ID_y  SITE_CODE  
SITE_ID                                                   
b47a998a3e        98      98    98         98         98  
a7bd4c027b        70      70    70         70         70  
5c565fbfb1        63      63    63         63         63  
7f76d66503        62      62    62         62         62  
585a3bc66c        54      54    54         54         54  
928a59b635        52      52    52         52         52  
c07a544c63        52      52    52         52         52  
8fd6f8d1c5        50      50    50         50         50  
bc58a52936        50      50    50         50         50  
91e86be9a6        50      50    50         50         50

In [162]:
#the 10 least used sites in the three days
siteid.tail(10)

PRODUCT  VALUE  DATE_TIME  DW_A_NUMBER  DW_B_NUMBER  CELL_ID_x  \
SITE_ID                                                                      
e47ea5e90b        1      1          1            1            1          1   
MtwILV82zT        1      1          1            1            1          1   
e26f7500ad        1      1          1            1            1          1   
e252847e66        1      1          1            1            1          1   
098f2f154a        1      1          1            1            1          1   
MW5J8MT3Zs        1      1          1            1            1          1   
c9cb7e8be0        1      1          1            1            1          1   
e1629ca4b7        1      1          1            1            1          1   
9f9bb90c1f        1      1          1            1            1          1   
KE4VqUMLIn        1      1          1            1            1          1   

            VILLES  STATUS  LOCALISATION  DECOUPZONE  ZONENAME  LONGITUDE  \
SITE_ID                                                                     
e47ea5e90b       1       1             1           1         1          1   
MtwILV82zT       1       1             1           1         1          1   
e26f7500ad       1       1             1           1         1          1   
e252847e66       1       1             1           1         1          1   
098f2f154a       1       1             1           1         1          1   
MW5J8MT3Zs       1       1             1           1         1          1   
c9cb7e8be0       1       1             1           1         1          1   
e1629ca4b7       1       1             1           1         1          1   
9f9bb90c1f       1       1             1           1         1          1   
KE4VqUMLIn       1       1             1           1         1          1   

            LATITUDE  REGION  AREA  CELL_ID_y  SITE_CODE  
SITE_ID                                                   
e47ea5e90b         1       1     1          1          1  
MtwILV82zT         1       1     1          1          1  
e26f7500ad         1       1     1          1          1  
e252847e66         1       1     1          1          1  
098f2f154a         1       1     1          1          1  
MW5J8MT3Zs         1       1     1          1          1  
c9cb7e8be0         1       1     1          1          1  
e1629ca4b7         1       1     1          1          1  
9f9bb90c1f         1       1     1          1          1  
KE4VqUMLIn         1       1     1          1          1

In [163]:
#grouping by time
time = alldata.groupby('DATE_TIME')
time2 =time.count()
time2

PRODUCT  VALUE  DW_A_NUMBER  DW_B_NUMBER  CELL_ID_x  \
DATE_TIME                                                                  
2012-05-06 23:11:27        4      4            4            4          4   
2012-05-06 23:12:07        2      2            2            2          2   
2012-05-06 23:12:10        2      2            2            2          2   
2012-05-06 23:12:25        4      4            4            4          4   
2012-05-06 23:12:36        2      2            2            2          2   
...                      ...    ...          ...          ...        ...   
2012-05-08 00:01:21       25     25           25           25         25   
2012-05-08 00:01:22       29     29           29           29         29   
2012-05-08 00:01:23       35     35           35           35         35   
2012-05-08 00:01:24       26     26           26           26         26   
2012-05-08 00:01:25        2      2            2            2          2   

                     SITE_ID  VILLES  STATUS  LOCALISATION  DECOUPZONE  \
DATE_TIME                                                                
2012-05-06 23:11:27        4       4       4             4           4   
2012-05-06 23:12:07        2       2       2             2           2   
2012-05-06 23:12:10        2       2       2             2           2   
2012-05-06 23:12:25        4       4       4             4           4   
2012-05-06 23:12:36        2       2       2             2           2   
...                      ...     ...     ...           ...         ...   
2012-05-08 00:01:21       25      25      25            25          25   
2012-05-08 00:01:22       29      29      29            29          29   
2012-05-08 00:01:23       35      35      35            35          35   
2012-05-08 00:01:24       26      26      26            26          26   
2012-05-08 00:01:25        2       2       2             2           2   

                     ZONENAME  LONGITUDE  LATITUDE  REGION  AREA  CELL_ID_y  \
DATE_TIME                                                                     
2012-05-06 23:11:27         4          4         4       4     4          4   
2012-05-06 23:12:07         2          2         2       2     2          2   
2012-05-06 23:12:10         2          2         2       2     2          2   
2012-05-06 23:12:25         4          4         4       4     4          4   
2012-05-06 23:12:36         2          2         2       2     2          2   
...                       ...        ...       ...     ...   ...        ...   
2012-05-08 00:01:21        25         25        25      25    25         25   
2012-05-08 00:01:22        29         29        29      29    29         29   
2012-05-08 00:01:23        35         35        35      35    35         35   
2012-05-08 00:01:24        26         26        26      26    26         26   
2012-05-08 00:01:25         2          2         2       2     2          2   

                     SITE_CODE  
DATE_TIME                       
2012-05-06 23:11:27          4  
2012-05-06 23:12:07          2  
2012-05-06 23:12:10          2  
2012-05-06 23:12:25          4  
2012-05-06 23:12:36          2  
...                        ...  
2012-05-08 00:01:21         25  
2012-05-08 00:01:22         29  
2012-05-08 00:01:23         35  
2012-05-08 00:01:24         26  
2012-05-08 00:01:25          2  

[1377 rows x 17 columns]

In [164]:
#preview of the top 200 time frames
timesort = time2.sort_values('PRODUCT',ascending = False)
timesort.head(200)

PRODUCT  VALUE  DW_A_NUMBER  DW_B_NUMBER  CELL_ID_x  \
DATE_TIME                                                                  
2012-05-07 00:00:05       92     92           92           92         92   
2012-05-07 00:00:06       86     86           86           86         86   
2012-05-07 00:00:20       84     84           84           84         84   
2012-05-07 00:00:09       84     84           84           84         84   
2012-05-07 00:00:07       82     82           82           82         82   
...                      ...    ...          ...          ...        ...   
2012-05-06 23:38:38        4      4            4            4          4   
2012-05-06 23:43:41        4      4            4            4          4   
2012-05-06 23:38:40        4      4            4            4          4   
2012-05-06 23:38:57        4      4            4            4          4   
2012-05-06 23:40:49        4      4            4            4          4   

                     SITE_ID  VILLES  STATUS  LOCALISATION  DECOUPZONE  \
DATE_TIME                                                                
2012-05-07 00:00:05       92      92      92            92          92   
2012-05-07 00:00:06       86      86      86            86          86   
2012-05-07 00:00:20       84      84      84            84          84   
2012-05-07 00:00:09       84      84      84            84          84   
2012-05-07 00:00:07       82      82      82            82          82   
...                      ...     ...     ...           ...         ...   
2012-05-06 23:38:38        4       4       4             4           4   
2012-05-06 23:43:41        4       4       4             4           4   
2012-05-06 23:38:40        4       4       4             4           4   
2012-05-06 23:38:57        4       4       4             4           4   
2012-05-06 23:40:49        4       4       4             4           4   

                     ZONENAME  LONGITUDE  LATITUDE  REGION  AREA  CELL_ID_y  \
DATE_TIME                                                                     
2012-05-07 00:00:05        92         92        92      92    92         92   
2012-05-07 00:00:06        86         86        86      86    86         86   
2012-05-07 00:00:20        84         84        84      84    84         84   
2012-05-07 00:00:09        84         84        84      84    84         84   
2012-05-07 00:00:07        82         82        82      82    82         82   
...                       ...        ...       ...     ...   ...        ...   
2012-05-06 23:38:38         4          4         4       4     4          4   
2012-05-06 23:43:41         4          4         4       4     4          4   
2012-05-06 23:38:40         4          4         4       4     4          4   
2012-05-06 23:38:57         4          4         4       4     4          4   
2012-05-06 23:40:49         4          4         4       4     4          4   

                     SITE_CODE  
DATE_TIME                       
2012-05-07 00:00:05         92  
2012-05-07 00:00:06         86  
2012-05-07 00:00:20         84  
2012-05-07 00:00:09         84  
2012-05-07 00:00:07         82  
...                        ...  
2012-05-06 23:38:38          4  
2012-05-06 23:43:41          4  
2012-05-06 23:38:40          4  
2012-05-06 23:38:57          4  
2012-05-06 23:40:49          4  

[200 rows x 17 columns]

In [165]:
#grouping by date
Date =alldata.resample('D', on = 'DATE_TIME')
Date

In [166]:
Datecount = Date.count()
Datecount

PRODUCT  VALUE  DATE_TIME  DW_A_NUMBER  DW_B_NUMBER  CELL_ID_x  \
DATE_TIME                                                                    
2012-05-06     1258   1258       1258         1258         1258       1258   
2012-05-07     6132   6132       6132         6132         6132       6132   
2012-05-08     2649   2649       2649         2649         2649       2649   

            SITE_ID  VILLES  STATUS  LOCALISATION  DECOUPZONE  ZONENAME  \
DATE_TIME                                                                 
2012-05-06     1258    1258    1258          1258        1258      1258   
2012-05-07     6132    6132    6132          6132        6132      6132   
2012-05-08     2649    2649    2649          2649        2649      2649   

            LONGITUDE  LATITUDE  REGION  AREA  CELL_ID_y  SITE_CODE  
DATE_TIME                                                            
2012-05-06       1258      1258    1258  1258       1258       1258  
2012-05-07       6132      6132    6132  6132       6132       6132  
2012-05-08       2649      2649    2649  2649       2649       2649

In [167]:
#grouping by time per hour
hour =alldata.resample('H', on = 'DATE_TIME')


In [168]:
#Number of entries recorded per hour
hourcount = hour.count()
hourcount

PRODUCT  VALUE  DATE_TIME  DW_A_NUMBER  DW_B_NUMBER  \
DATE_TIME                                                                  
2012-05-06 23:00:00     1258   1258       1258         1258         1258   
2012-05-07 00:00:00     5400   5400       5400         5400         5400   
2012-05-07 01:00:00        0      0          0            0            0   
2012-05-07 02:00:00        0      0          0            0            0   
2012-05-07 03:00:00        0      0          0            0            0   
2012-05-07 04:00:00        0      0          0            0            0   
2012-05-07 05:00:00        0      0          0            0            0   
2012-05-07 06:00:00        0      0          0            0            0   
2012-05-07 07:00:00        0      0          0            0            0   
2012-05-07 08:00:00        0      0          0            0            0   
2012-05-07 09:00:00        0      0          0            0            0   
2012-05-07 10:00:00        0      0          0            0            0   
2012-05-07 11:00:00        0      0          0            0            0   
2012-05-07 12:00:00        0      0          0            0            0   
2012-05-07 13:00:00        0      0          0            0            0   
2012-05-07 14:00:00        0      0          0            0            0   
2012-05-07 15:00:00        0      0          0            0            0   
2012-05-07 16:00:00        0      0          0            0            0   
2012-05-07 17:00:00        0      0          0            0            0   
2012-05-07 18:00:00        0      0          0            0            0   
2012-05-07 19:00:00        0      0          0            0            0   
2012-05-07 20:00:00        0      0          0            0            0   
2012-05-07 21:00:00        0      0          0            0            0   
2012-05-07 22:00:00        0      0          0            0            0   
2012-05-07 23:00:00      732    732        732          732          732   
2012-05-08 00:00:00     2649   2649       2649         2649         2649   

                     CELL_ID_x  SITE_ID  VILLES  STATUS  LOCALISATION  \
DATE_TIME                                                               
2012-05-06 23:00:00       1258     1258    1258    1258          1258   
2012-05-07 00:00:00       5400     5400    5400    5400          5400   
2012-05-07 01:00:00          0        0       0       0             0   
2012-05-07 02:00:00          0        0       0       0             0   
2012-05-07 03:00:00          0        0       0       0             0   
2012-05-07 04:00:00          0        0       0       0             0   
2012-05-07 05:00:00          0        0       0       0             0   
2012-05-07 06:00:00          0        0       0       0             0   
2012-05-07 07:00:00          0        0       0       0             0   
2012-05-07 08:00:00          0        0       0       0             0   
2012-05-07 09:00:00          0        0       0       0             0   
2012-05-07 10:00:00          0        0       0       0             0   
2012-05-07 11:00:00          0        0       0       0             0   
2012-05-07 12:00:00          0        0       0       0             0   
2012-05-07 13:00:00          0        0       0       0             0   
2012-05-07 14:00:00          0        0       0       0             0   
2012-05-07 15:00:00          0        0       0       0             0   
2012-05-07 16:00:00          0        0       0       0             0   
2012-05-07 17:00:00          0        0       0       0             0   
2012-05-07 18:00:00          0        0       0       0             0   
2012-05-07 19:00:00          0        0       0       0             0   
2012-05-07 20:00:00          0        0       0       0             0   
2012-05-07 21:00:00          0        0       0       0             0   
2012-05-07 22:00:00          0        0       0       0            

EVALUATION

This analysis yeilded the following information that should be considered in the infrastructure upgrading strategy.
1. The most popular product used in the 3 days was voice followed by sms and then data.
   The total value(from billing information) gotten from each product was:
        Voice - 247,243
        sms - 31,883
        data - 1,809
2. In terms of localisation about 41.72% of the data recorded was from Abidjan and the Interior recorded 
   about 58.27% of the data.
3. All the data was recorded within 3 days.
4. Of the 3days the 7th of May recorded the most entries which were 6,132 followed by 8th of May with 2,649
   and 6th May with 1,258
5. Most of the data recorded was between the hours of 2300HRS and midnight.